node/src/ethereum.rs

```diff
@@ -0,0 +1,347 @@
+{
+    use fc_rpc::{
+        pending::AuraConsensusDataProvider, Debug, DebugApiServer, Eth, EthApiServer, EthDevSigner,
+        EthFilter, EthFilterApiServer, EthPubSub, EthPubSubApiServer, EthSigner, Net, NetApiServer,
+        Web3, Web3ApiServer,
+    };
+    #[cfg(feature = "txpool")]
+    use fc_rpc::{TxPool, TxPoolApiServer};
+
+    let EthDeps {
+        client,
+        pool,
+        graph,
+        converter,
+        is_authority,
+        enable_dev_signer,
+        network,
+        sync,
+        frontier_backend,
+        storage_override,
+        block_data_cache,
+        filter_pool,
+        max_past_logs,
+        fee_history_cache,
+        fee_history_cache_limit,
+        execute_gas_limit_multiplier,
+        forced_parent_hashes,
+        pending_create_inherent_data_providers,
+    } = deps;
+
+    let mut signers = Vec::new();
+    if enable_dev_signer {
+        signers.push(Box::new(EthDevSigner::new()) as Box<dyn EthSigner>);
+    }
+
+    io.merge(
+        Eth::<B, C, P, CT, BE, A, CIDP, EC>::new(
+            client.clone(),
+            pool.clone(),
+            graph.clone(),
+            converter,
+            sync.clone(),
+            signers,
+            storage_override.clone(),
+            frontier_backend.clone(),
+            is_authority,
+            block_data_cache.clone(),
+            fee_history_cache,
+            fee_history_cache_limit,
+            execute_gas_limit_multiplier,
+            forced_parent_hashes,
+            pending_create_inherent_data_providers,
+            Some(Box::new(AuraConsensusDataProvider::new(client.clone()))),
+        )
+        .replace_config::<EC>()
+        .into_rpc(),
+    )?;
+
+    if let Some(filter_pool) = filter_pool {
+        io.merge(
+            EthFilter::new(
+                client.clone(),
+                frontier_backend.clone(),
+                graph.clone(),
+                filter_pool,
+                500_usize, // max stored filters
+                max_past_logs,
+                block_data_cache.clone(),
+            )
+            .into_rpc(),
+        )?;
+    }
+
+    io.merge(
+        EthPubSub::new(
+            pool,
+            client.clone(),
+            sync,
+            subscription_task_executor,
+            storage_override.clone(),
+            pubsub_notification_sinks,
+        )
+        .into_rpc(),
+    )?;
+
+    io.merge(
+        Net::new(
+            client.clone(),
+            network,
+            // Whether to format the `peer_count` response as Hex (default) or not.
+            true,
+        )
+        .into_rpc(),
+    )?;
+
+    io.merge(Web3::new(client.clone()).into_rpc())?;
+
+    io.merge(
+        Debug::new(
+            client.clone(),
+            frontier_backend,
+            storage_override,
+            block_data_cache,
+        )
+        .into_rpc(),
+    )?;
+
+    #[cfg(feature = "txpool")]
+    io.merge(TxPool::new(client, graph).into_rpc())?;
+
+    Ok(io)
+}
```

Comment:
I recommend breaking this up into helper functions, especially if we think this will grow to be even longer

Reply:
Done, please review @sam0x17